src/extern/turbomole.f90

```diff
@@ -163,58 +174,126 @@ subroutine singlepoint(self, env, mol, chk, printlevel, restart, &
+   real(wp), intent(in) :: chrg
+      !! charge
+   integer, intent(in) :: uhf
+      !! multiplicity, number of unpaired electrons
+
    character(len=255) atmp
    character(len=:), allocatable :: syspath, cefine
-   logical :: cache, exist
+   logical :: cache
+      !! to check if molecule moves b/n optimization runs
+   logical :: exist
+   logical :: runCefine
+   integer :: err
 
    cache = .false.
    dip=0
-
+   
+   !> TM input
    inquire(file="control", exist=exist)
+      !! if control file present in the calc dir
    if (.not.exist) then
-      call rdvar("PATH", syspath)
-      call rdpath(syspath, "cefine", cefine, exist)
-      if (exist) then
-         call wrtm(n,at,xyz)
-         call execute_command_line("exec "//cefine//" --func tpss --def2/SVP --cosmo 2.38 --d4 -sym c1 --noopt")
-      end if
+      
+      !> cefine 
+      if(len(set%ext_turbo%input_string).ne.0) then      
+         call rdvar("PATH", syspath)
+         call rdpath(syspath, "cefine", cefine, exist)
+         if (exist) then
+            call wrtm(n,at,xyz)
+            if (allocated(set%ext_turbo%input_string)) then 
+               call execute_command_line("exec "//cefine//" "//set%ext_turbo%input_string)
+            else
+               call execute_command_line("exec "//cefine//" --func tpss --bas def2-SVP --d4 -sym c1 --noopt")
+            endif
+         end if
+
+      !> default control file
+      else
+         call writeControl(chrg,uhf)
+      endif
+
    end if
 
+   inquire(file="control", exist=exist)
    if (.not.exist) then
       call env%error("No 'control' file in current directory")
       return
    end if
-
+   
+  
    ! TM (RI)
    if(extcode.eq.1)then
       !$omp critical (turbo_lock)
       inquire(file='gradient', exist=exist)
       if (exist .and. grd) then
-         call rdtm(n,grd,eel,g,xyz_cached)
-         cache = all(abs(xyz_cached - xyz) < 1.e-10_wp)
+         call rdtm(env,n,grd,eel,g,xyz_cached)
+         cache = all(abs(xyz_cached - xyz) < 1.e-7_wp)
       end if
       if (.not.cache) then
          call wrtm(n,at,xyz)
          if(extmode.eq.1)then
-            call execute_command_line('exec ridft  >  job.last 2>> /dev/null')
-            if(grd)call execute_command_line('exec rdgrad >> job.last 2>> /dev/null ')
-         endif
+            
+            write(env%unit,'(72("="))')
+            write(env%unit,'(1x,"*",1x,a)') &
+               "letting TURBOMOLE take over the control..."
+            
+            if (set%oniom_settings%silent) then
+               call execute_command_line('exec ridft >  job.last 2>> /dev/null',exitstat=err)
+               if(grd)call execute_command_line('exec rdgrad >> job.last 2>> /dev/null ',exitstat=err)
+            else
+               call execute_command_line('exec ridft  | tee  job.last 2>> /dev/null',exitstat=err)
+               if(grd)call execute_command_line('exec rdgrad |tee -a job.last 2>> /dev/null ',exitstat=err)
+            endif
+
```

Comment:
Same thing here - we should check if the user has Turbomole (``ridft``) before using it. Or is this done somewhere else already?